rust-runtime/aws-smithy-runtime/src/expiring_cache.rs

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::future::Future;
```

Comment:
do we want this to be in runtime-api instead? I think we've hit other issues where this primitive wasn't accessible to the dependency tree.

Reply:
It looks like `aws-config` is the only other place that uses it, and that depends on `aws-smithy-runtime`. So I think it's fine where it is. We can always move/re-export it later if we need to.